tests/gamma/csm_observability_test.py

```diff
@@ -25,13 +26,77 @@
 logger = logging.getLogger(__name__)
 flags.adopt_module_key_flags(xds_k8s_testcase)
 
+# Type aliases
 _XdsTestServer = xds_k8s_testcase.XdsTestServer
 _XdsTestClient = xds_k8s_testcase.XdsTestClient
 _Lang = skips.Lang
 
+# Testing consts
+TEST_RUN_SECS = 90
+REQUEST_PAYLOAD_SIZE = 271828
+RESPONSE_PAYLOAD_SIZE = 314159
+VALUE_NOT_CHECKED = object()  # Sentinel value
+GRPC_METHOD_NAME = "grpc.testing.TestService/UnaryCall"
+CSM_WORKLOAD_NAME_SERVER = "csm_workload_name_from_server"
+CSM_WORKLOAD_NAME_CLIENT = "csm_workload_name_from_client"
+CSM_CANONICAL_SERVICE_NAME_SERVER = "csm_canonical_service_name_from_server"
+CSM_CANONICAL_SERVICE_NAME_CLIENT = "csm_canonical_service_name_from_client"
+PROMETHEUS_HOST = "prometheus.googleapis.com"
+METRIC_CLIENT_ATTEMPT_SENT = (
+    f"{PROMETHEUS_HOST}/"
+    "grpc_client_attempt_sent_total_compressed_message_size_bytes/histogram"
+)
+METRIC_CLIENT_ATTEMPT_RCVD = (
+    f"{PROMETHEUS_HOST}/"
+    "grpc_client_attempt_rcvd_total_compressed_message_size_bytes/histogram"
+)
+METRIC_CLIENT_ATTEMPT_DURATION = (
+    f"{PROMETHEUS_HOST}/grpc_client_attempt_duration_seconds/histogram"
+)
+METRIC_CLIENT_ATTEMPT_STARTED = (
+    f"{PROMETHEUS_HOST}/grpc_client_attempt_started_total/counter"
+)
+METRIC_SERVER_CALL_RCVD = (
+    f"{PROMETHEUS_HOST}/"
+    "grpc_server_call_rcvd_total_compressed_message_size_bytes/histogram"
+)
+METRIC_SERVER_CALL_SENT = (
+    f"{PROMETHEUS_HOST}/"
+    "grpc_server_call_sent_total_compressed_message_size_bytes/histogram"
+)
+METRIC_SERVER_CALL_DURATION = (
+    f"{PROMETHEUS_HOST}/grpc_server_call_duration_seconds/histogram"
+)
+METRIC_SERVER_CALL_STARTED = (
+    f"{PROMETHEUS_HOST}/grpc_server_call_started_total/counter"
+)
+HISTOGRAM_CLIENT_METRICS = [
+    METRIC_CLIENT_ATTEMPT_SENT,
+    METRIC_CLIENT_ATTEMPT_RCVD,
+    METRIC_CLIENT_ATTEMPT_DURATION,
+]
+HISTOGRAM_SERVER_METRICS = [
+    METRIC_SERVER_CALL_DURATION,
+    METRIC_SERVER_CALL_RCVD,
+    METRIC_SERVER_CALL_SENT,
+]
+COUNTER_CLIENT_METRICS = [
+    METRIC_CLIENT_ATTEMPT_STARTED,
+]
+COUNTER_SERVER_METRICS = [
+    METRIC_SERVER_CALL_STARTED,
+]
+HISTOGRAM_METRICS = HISTOGRAM_CLIENT_METRICS + HISTOGRAM_SERVER_METRICS
+COUNTER_METRICS = COUNTER_CLIENT_METRICS + COUNTER_SERVER_METRICS
+CLIENT_METRICS = HISTOGRAM_CLIENT_METRICS + COUNTER_CLIENT_METRICS
+SERVER_METRICS = HISTOGRAM_SERVER_METRICS + COUNTER_SERVER_METRICS
+ALL_METRICS = HISTOGRAM_METRICS + COUNTER_METRICS
+
```

Comment:
I think we can make it even easier to understand with a dataclass to describe metric

```py
@dataclasses.dataclass(eq=False)
class Metric:
    name: str
    resource_labels: dict[str, str]
    metric_labels: dict[str, str]
    points: list[gm_metric.Point]

    @classmethod
    def from_response(
        cls,
        name: str,
        response: gm_metric.TimeSeries,
    ) -> "Metric":
        return cls(
            name=name,
            resource_labels=dict(response.metric.metric.labels),
            metric_labels=dict(response.metric.resource.labels),
            points=list(response.points),
        )
```

Let's discuss it IRL

Reply:
Working on this.

Reply:
Just FYI: I wasn't 100% sure this structure works for you, so I wanted to discuss this IRL first.

Reply:
Added the `MetricTimeSeries` `dataclass` here. Please review. Thanks. It has a `pretty_print()` method. In the rest of the test, when we access `all_results[metric]` we know we are getting a `MetricTimeSeries` back.